addons/website/static/src/scss/website.scss

```diff
@@ -314,13 +346,75 @@ h5, h6 {
 h6 {
     color: map-get($colors, 'o-cc1-h6');
 }
+
+$heading-font-families: () !default;
+$heading-font-families: map-merge((
+    "h2": $h2-font-family,
+    "h3": $h3-font-family,
+    "h4": $h4-font-family,
+    "h5": $h5-font-family,
+    "h6": $h6-font-family,
+), $heading-font-families);
+$heading-line-heights: () !default;
+$heading-line-heights: map-merge((
+    "h2": $h2-line-height,
+    "h3": $h3-line-height,
+    "h4": $h4-line-height,
+    "h5": $h5-line-height,
+    "h6": $h6-line-height,
+), $heading-line-heights);
+$heading-margin-tops: () !default;
+$heading-margin-tops: map-merge((
+    "h2": $h2-margin-top,
+    "h3": $h3-margin-top,
+    "h4": $h4-margin-top,
+    "h5": $h5-margin-top,
+    "h6": $h6-margin-top,
+), $heading-margin-tops);
+$heading-margin-bottoms: () !default;
+$heading-margin-bottoms: map-merge((
+    "h2": $h2-margin-bottom,
+    "h3": $h3-margin-bottom,
+    "h4": $h4-margin-bottom,
+    "h5": $h5-margin-bottom,
+    "h6": $h6-margin-bottom,
+), $heading-margin-bottoms);
+
+h1, .h1, h2, .h2, h3, .h3, h4, .h4, h5, .h5, h6, .h6 {
+    line-height: $headings-line-height;
+    margin-top: $headings-margin-top;
+    margin-bottom: $headings-margin-bottom;
+}
+@for $index from 2 through 6 {
+    h#{$index}, .h#{$index} {
+        $hn: 'h#{$index}';
+        @if ($headings-font-family != map-get($heading-font-families, $hn)) {
+            font-family: map-get($heading-font-families, $hn);
+        }
+        @if ($headings-line-height != map-get($heading-line-heights, $hn)) {
+            line-height: map-get($heading-line-heights, $hn);
+        }
+        @if ($headings-margin-top != map-get($heading-margin-tops, $hn)) {
+            margin-top: map-get($heading-margin-tops, $hn);
+        }
+        @if ($headings-margin-bottom != map-get($heading-margin-bottoms, $hn)) {
+            margin-bottom: map-get($heading-margin-bottoms, $hn);
+        }
+    }
+}
+
 .btn {
     @if ($o-theme-buttons-font != $o-theme-font) {
         font-family: $o-theme-buttons-font;
     }
 }
 
 // Texts
+p {
+    margin-top: $paragraph-margin-top;
+    margin-bottom: $paragraph-margin-bottom;
```

Comment:
Why not removing the second rule (margin-bottom) which is a pure duplicate of the bootstrap one?